restricted_site_access.php

```diff
@@ -513,6 +515,20 @@ public static function admin_notice() {
 	}
 
 	/**
+	 * Check if the page caching is on, and notify the admin
+	 */
+	public static function page_cache_notice() {
+		//If WP_CACHE is on we show notification
+		if ( true === WP_CACHE ) {
+			$notif_message = __( 'You have page caching activated. Restricted Site Access plugin may not working as expected for cached page.', 'restricted-site-access' );
+		}
+
+		if ( isset( $notif_message ) ) {
+			echo '<div class="error"><p><strong>' . $notif_message . '</strong></p></div>';
```

Comment:
Needs escaping on the message. `esc_html()` would probably suffice. Note that `__()` is not escaped by default.

See: https://10up.github.io/Engineering-Best-Practices/php/#escape-or-validate-output